samples/net/echo_server/overlay-ot.conf

```diff
@@ -0,0 +1,28 @@
+CONFIG_CPLUSPLUS=y
+
+CONFIG_NET_IPV4=n
+CONFIG_NET_IPV6_NBR_CACHE=n
+CONFIG_NET_IPV6_MLD=n
+CONFIG_NET_APP_NEED_IPV6=y
+CONFIG_NET_APP_NEED_IPV4=n
+
+CONFIG_MAIN_STACK_SIZE=2048
+CONFIG_SYSTEM_WORKQUEUE_STACK_SIZE=2048
+CONFIG_NET_TX_STACK_SIZE=2048
+CONFIG_NET_RX_STACK_SIZE=3072
+
+CONFIG_NET_L2_IEEE802154=n
+CONFIG_NET_L2_OPENTHREAD=y
+
+CONFIG_OPENTHREAD_DEBUG=y
+CONFIG_OPENTHREAD_L2_DEBUG=y
+CONFIG_OPENTHREAD_L2_LOG_LEVEL_INFO=y
+CONFIG_OPENTHREAD_LOG_LEVEL_INFO=y
+
+CONFIG_OPENTHREAD_CHANNEL=26
+
+# We need a bit more buffers for OT
+CONFIG_NET_PKT_RX_COUNT=50
+CONFIG_NET_PKT_TX_COUNT=50
+CONFIG_NET_BUF_RX_COUNT=50
+CONFIG_NET_BUF_TX_COUNT=50
```

Comment:
We should specifiy IPv6 addresses for Thread explicitly:
```
CONFIG_NET_APP_MY_IPV6_ADDR="fdde:ad00:beef::2"
CONFIG_NET_APP_PEER_IPV6_ADDR="fdde:ad00:beef::1"
```
I saw that kw41z didn't do it in it's config for OT, yet it's not correct even if it worked with default addresses used by the sample. Thread devices should not add addresses with prefixes that are not registered in the network, it's a spec violation.